### ISSUE TITLE
Identify minimum viable config docs for standalone Elastic Agent

### ISSUE THREAD

**Issue by dedemorton** [elastic/observability-docs#1905]
Users running standalone agents need reference documentation that describes:

- The structure of the standalone config file, elastic-agent.yml.
- The settings that are valid in the configuration and their defaults.
- Realist configuration examples.

We need to identify the minimal config documentation that standalone users require to be successful.

### Assumptions (draft)
- Standalone users will use Kibana to configure integrations and generate the initial configuration file. After the initial file is generated, users may want to modify settings in the configuration file. They should be able to access reference documentation about valid settings rather than going back into the UI or trying to figure out what’s possible by looking at the old Beats docs.

- We don’t want standalone users to refer to the existing Filebeat and Metricbeat settings documentation because it contains settings that Elastic Agent does not support, and we don’t want users to mistakenly attempt to modify the configs for managed Beats.
 
### Open questions
**Q:** Who is the target audience for this documentation? Just standalone users? Will Fleet users what to refer to reference docs for extended descriptions about settings that surface in the config UI?

**Q:** Who will own and maintain this content over time? Writing team? Development team? PMs?

### What content do we have right now?
The settings reference documentation currently provides _detailed_ reference content about:

[Providers](https://www.elastic.co/guide/en/fleet/current/providers.html)
[Outputs](https://www.elastic.co/guide/en/fleet/current/elastic-agent-output-configuration.html)
[Logging](https://www.elastic.co/guide/en/fleet/current/elastic-agent-standalone-logging-config.html)
[Monitoring](https://www.elastic.co/guide/en/fleet/current/elastic-agent-monitoring-configuration.html)

### What content is missing?

Inputs (includes logfile and metric inputs)
Processors
Agent settings (download, process, grpc, retry, reload

### What content is in-progress?

https://github.com/elastic/observability-docs/pull/500 - This content is very stale (based on Beats docs from 2021). It is missing some processors and other content might be out-of-date. Some of the content has been edited.

https://github.com/elastic/observability-docs/pull/424 - This content was updated recently and some of the content has been edited. It contains a lot of open questions and includes many settings we probably do not want to expose. It also covers ancillary info we will probably want to remove.

### Related issues

https://github.com/elastic/observability-docs/issues/1044  - Note that the support team and internal folks have been asking for more detailed documentation for standalone users for a long time.

**Comment by mbarretta** [comment]
the gaps identified here are popping up more and more in the field - expanding the Agent docs to include these missing areas will ease a lot of pain on our users and ourselves

**Comment by dedemorton** [comment]
@ollyhowell I agree with Mike. Back in May, I sent out a request in email and slack asking for someone on the dev team to partner with me on these updates, but the response was 🦗 🦗 🦗.  We need someone on the dev team to help identify which settings are valid for agent, which settings we want to expose, etc.

**Comment by ollyhowell** [comment]
Spoke with @jlind23 briefly and he's going to put this into their 8.6 cycle and see how much bandwidth they can dedicate

**Comment by mukeshelastic** [comment]
@jlind23 do we have a dev issue to track this work in our sprints? If not, let's create one and if yes and it isn't prioritized, let's discuss in sprint planning for prioritization

**Comment by jlind23** [comment]
@mukeshelastic we should rather keep the same issue and assign someone to pair with dede on this.
I raised this to @pierrehilbert and pre-scoped it for our next sprint

**Comment by dedemorton** [comment]
@jlind23 @mukeshelastic I've revived the #500 and updated it with the latest changes from the Beats repo, but it's still a work-in-progress. I need someone to look at the processors I've listed in the PR description and let me know which processors we want to document (that is, which processors will actually work with Elastic Agent). Who can help me with this?

I'll continue with some of the improvements I'm planning and focus on the info that's likely to stay, but would appreciate some help with this soon.

**Comment by jlind23** [comment]
@nimarezainia and @pierrehilbert would be best here to help you out or find someone able to.
